sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/CatalogManager.scala

```diff
@@ -77,40 +77,37 @@ class CatalogManager(conf: SQLConf, defaultSessionCatalog: TableCatalog) extends
 
   private def getDefaultNamespace(c: CatalogPlugin) = c match {
     case c: SupportsNamespaces => c.defaultNamespace()
-    case _ => Array.empty[String]
+    case _ => Array("default") // The builtin catalog use "default" as the default database.
```

Comment:
If a catalog doesn't implememt `SupportsNamespaces`, I think it's more reasonable to use empty namespace as the default.

For the builtin catalog, can we implement the `defaultNamespace` in `V2SessionCatalog`?

Reply:
Thanks @cloud-fan. I will handle this as you suggested.

Reply:
Now this will return `Array.empty[String]` if a catalog doesn't support namespaces.

Also, instead of `CatalogExtension` extending `SupportsNamespace`, I introduced `BaseSessionCatalog` which inherits both `TableCatalog` and `SupportsNamespace`, and `CatalogExtension` extends `BaseSessionCatalog`.

This is so that any session catalog can extend `BaseSessionCatalog` instead of `CatalogExtension`, which requires overriding `setDelegateCatalog`.